Sprint-1/exercises/paths.js

```diff
@@ -9,10 +9,15 @@
 
 // (All spaces in the "" line should be ignored. They are purely for formatting.)
 
-const filePath = "/Users/mitch/cyf/Module-JS1/week-1/interpret/file.txt";
+const filePath = "/Users/mitch/cyf/Module-JS1/week-1/interpret/file.db";
 const lastSlashIndex = filePath.lastIndexOf("/");
 const base = filePath.slice(lastSlashIndex + 1);
 console.log(`The base part of ${filePath} is ${base}`);
 
 // Create a variable to store the dir part of the filePath variable
+const dirPart = filePath.slice(0, lastSlashIndex)
+console.log(`The dir part of ${filePath} is ====> ${dirPart}`);
 // Create a variable to store the ext part of the variable
```

Comment:
This works, good job!

What do you think would happen if the filename didn't contain a `.`?

What do you think _should_ happen if the filename doesn't contain a `.`?

Reply:
```lastDotIndex``` would be -1  and ```extPart``` would return whatever the base is.

Reply:
Yeah, that's what _would_ happen. What do you think _should_ happen? If I have a file just named `image`, and ask you what its extension is, what do you think you'd say?

Reply:
if the file is of type image the extension would be either svg, png, jpeg etc... I am not sure I get the question right but that's what I would say

Reply:
But I'm asking: what if the file _doesn't_ have an extension?

I can create a file on my computer just named `image` that _doesn't_ have an extension.

I agree that most files _should_ have an extension, but there is nothing that _guarantees_ that. What do you think should happen if we are given a file name that _doesn't_ match our expectations?

Reply:
In this case the exPart variable will evaluate to ```image```.

Reply:
Is that what we _want_ though? If I asked you as a human, rather than code, "What is the extension of this file name", what would you answer?

Reply:
I would say it does not have an extension.

Reply:
Cool - what value do you think we could return from this function so that the code does the same as you would as a human? (I can think of three better values than "image"!)

Reply:
I think we could return an error message stating that the file does not have an extension or we could just return an empty string

Reply:
An empty string I think makes sense. One disadvantage of this is that it means that getting the extension of a file named "image" or a file named "image." would give the same result - it is sometimes useful to know the difference.

I think `null` ("There was no extension") would also be a valid thing to return (compared with the empty string, when there was a `.` but nothing after it).

An error may also make sense, but we wouldn't want to _return_ an error message (because if we just return a string, we'll assume that string was the extension!) - have a read of https://medium.com/@dilarauluturhan/javascript-error-handling-try-catch-throw-finally-de5d26f16790 to learn about error handling :)

Reply:
Great! the artilce is very helpful. Thank you.